packages/react-components/react-tree/library/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -145,10 +148,57 @@ export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDi
     });
   });
 
+  const { findFirstFocusable } = useFocusFinders();
+
+  const handleTreeGridActionsKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (!event.currentTarget.contains(event.target as Node) || !treeItemRef.current) {
+      return;
+    }
+    switch (event.key) {
+      case treeDataTypes.ArrowLeft: {
+        // TODO: this should be included in requestTreeResponse navigation signature
+        treeItemRef.current?.focus();
+        return;
+      }
```

Comment:
I don't  understand this TODO, does it mean that `requestTreeResponse` should actually handle navigation in/out of the actions?

Reply:
It does sound like the right thing to do here IMO, I'd expect all navigation on the tree to be handled by the navigation pattern, that way we can inform the user on the `onNavigation` handler that some navigation to somewhere is happening. It does poses the fact that the navigation is to the same item, but still I don't see why not.

Reply:
If this should be the case we should just do it now no? What's the reason to keep this as a TODO?

Reply:
![image](https://github.com/user-attachments/assets/5d6b45d2-ea76-4210-9a0d-2aa411d3222c)

Reply:
Jokes aside, removed the `TODO` sign and reorganized the logic properly now 😉